lib/cwd.js

```diff
@@ -0,0 +1,49 @@
+import {statSync} from 'node:fs';
+import {resolve} from 'node:path';
+import {fileURLToPath} from 'node:url';
+import process from 'node:process';
+
+export const getDefaultCwd = () => {
+	try {
+		return process.cwd();
+	} catch (error) {
+		error.message = `The current directory does not exist.\n${error.message}`;
+		throw error;
+	}
+};
+
+export const normalizeCwd = cwd => {
+	const cwdString = safeNormalizeFileUrl(cwd, 'The "cwd" option');
+	return resolve(cwdString);
+};
+
+export const safeNormalizeFileUrl = (file, name) => {
+	const fileString = normalizeFileUrl(file);
+
+	if (typeof fileString !== 'string') {
+		throw new TypeError(`${name} must be a string or a file URL: ${fileString}.`);
+	}
+
+	return fileString;
+};
+
+export const normalizeFileUrl = file => file instanceof URL ? fileURLToPath(file) : file;
+
+export const fixCwdError = (originalMessage, cwd) => {
```

Comment:
We cannot perform this `cwd` validation check before process spawning because it involves making a `stat` I/O call. Right now, `execa()` spawns processes right away, and introducing a sync I/O call before spawning for every `execa()` call is not worth it, just to improve the `cwd` validation error message.

That being said, since `process.cwd()` is synchronous, when the `cwd` option default value is used, it is validated before process spawning.